framework/src/org/checkerframework/common/value/ValueTransfer.java

```diff
@@ -805,36 +804,49 @@ private Range calculateRangeUnaryOp(
             TransferInput<CFValue, CFStore> p,
             CFStore thenStore,
             CFStore elseStore) {
-        if (isIntRange(leftNode, p)
-                || isIntRange(rightNode, p)
-                || isIntegralUnknownVal(rightNode, p)
-                || isIntegralUnknownVal(leftNode, p)) {
+        return calculateBinaryComparison(
```

Comment:
What's the reasoning for keeping this version of `calculateBinaryComparison`? Why can't its callers just use the version below and call `getValueOfSubnode` themselves?